firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -185,7 +190,10 @@ private Map<DocumentKey, OverlayedDocument> computeViews(
     for (Map.Entry<DocumentKey, MutableDocument> entry : docs.entrySet()) {
       result.put(
           entry.getKey(),
-          new OverlayedDocument(entry.getValue(), mutatedFields.get(entry.getKey())));
+          new OverlayedDocument(
+              entry.getValue(),
+              keysWithOverlay.contains(entry.getKey()),
```

Comment:
the only reason we need `keysWithOverlay` is because of the `recalculateAndSaveOverlays` above, right? Otherwise we could just use `mutatedFields.contains(entry.getKey())`?

It makes me wonder about the discrepancy between `mutatedFields` and `keysWithOverlays`. Let's discuss it offline.

Reply:
I never thought of it actually, I think it is reasonable. FWIW, short of a SUM type solution, I am pretty indifferent to exactly how we will address this. It's all a hack.